### PR TITLE
Function to Disable & Remove Traffic Lights

### LINKED ISSUE #320: Button to remove / disable all traffic lights
Is it possible to implement a button that removes all traffic lights in your entire city? 

I tried implementing it myself but I can't figure out how to iterate the "ref NetNode node" from 0 to max and have it in a nested loop with 
`for (ushort nodeId = 0; nodeId < NetManager.MAX_NODE_COUNT; ++nodeId)`

### PR DESCRIPTION
**Pull request by krzychu124**
Fixes #320

Description:
- disable traffic lights at new junctions
  - touches only new junctions, not existing ones
  - still allows adding lights manually -- it only prevents lights at new junctions
- remove all traffic lights at existing junctions
  - doesn't remove timed traffic lights
  - because quite drastic and undoable: confirmation dialog before execution

Additional work done:
- moved implementation made by @Craxy from closed PR(#390).  
- changed manual detour to harmony postfix,
- adapted translations to new translation system ~~(new keys have to be added to Crowdin)~~ Done

### CODE REVIEW

**Comment by aubergine10** [comment]
Looks like it will need translations adding to options.csv for following keys:

**Policies tab:**

* `VR.Checkbox:Disable setting up traffic lights for new junctions` -- @VictorPhilipp once mentioned to me it's best to avoid 'negative' options, so this might be better as `VR.Checkbox:Automatically add traffic lights if applicable` (and default the setting to true)?

**Maintenance tab:**

* `Maintenance.Button:Remove all existing traffic lights`

Not sure about following keys - @kvakvs any suggestions?

* `Maintenance.Dialog.Title:Remove all traffic lights`
* `Maintenance.Dialog.Text:Remove all traffic lights, Confirmation`

**Comment by krzychu124** [comment]
Right, we have to add those keys. Text is not final because I wasn't sure how it should look like.
I forgot about `negative` option suggestion. I'd got 'false' as default value in my mind while I wrote that checkbox key text 😅  I like your suggested description.

File: TLM/TLM/Manager/Impl/OptionsManager.cs [hunk 1]
```diff
@@ -193,6 +193,10 @@ public class OptionsManager
                 OptionsVehicleRestrictionsTab.SetAllowFarTurnOnRed(data[40] == 1);
             }
 
+            if (data.Length >= 42) {
```
Comment (kvakvs) [inline TLM/TLM/Manager/Impl/OptionsManager.cs]: Magic numbers everywhere, can make enums or constants?

File: TLM/TLM/Manager/Impl/TrafficLightManager.cs [hunk 2]
```diff
@@ -108,6 +108,13 @@ public class TrafficLightManager
             return SetTrafficLight(nodeId, false, ref node, out reason);
         }
 
+        public void RemoveAllExistingTrafficLights() {
+            Array16<NetNode> nodes = NetManager.instance.m_nodes;
+            for (ushort i = 0; i < nodes.m_size; i++) {
+                RemoveTrafficLight(i, ref nodes.m_buffer[i]);
```
Comment (kvakvs) [inline TLM/TLM/Manager/Impl/TrafficLightManager.cs]: I had a very quick look in the code it seems there isn't memory moved on each traffic light removal, so should be fine. If this ever becomes slow, we can optimize it. Just keep an eye on performance here.

Comment (krzychu124) [reply TLM/TLM/Manager/Impl/TrafficLightManager.cs]: I will check performance of that method later :) 
This function will be triggered from mod options maintenance tab on demand, after user confirm this operation.

**Comment by krzychu124** [comment]
I've just benchmarked Remove traffic lights call and I must say it's not that bad. On my potato laptop, 30-35 ms, savegame with 650k population city

**Pull request merged**